src/core/trtserver.h

```diff
@@ -465,13 +466,103 @@ TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_TraceManagerNew(
+    int64_t model_version);
+
+/// Set the ID for the request in a request options. The response of the request
+/// will contain the same ID. The request sender can use the ID to correlate
+/// the response to corresponding request if needed.
+/// \param request_options The request options object.
+/// \param id The ID.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestOptionsSetId(
+    TRTSERVER_InferenceRequestOptions* request_options, uint64_t id);
+
+/// Set the flag associated with the request in a request options. 'flags'
+/// should holds a bitwise-or of all flag values. \param request_options The
+/// request options object. \param flags The flags. \return a TRTSERVER_Error
+/// indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestOptionsSetFlags(
+    TRTSERVER_InferenceRequestOptions* request_options, uint32_t flags);
+
+/// Set the correlation ID for the request in a request options.
+/// \param request_options The request options object.
+/// \param correlation_id The correlation ID.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error*
+TRTSERVER_InferenceRequestOptionsSetCorrelationId(
+    TRTSERVER_InferenceRequestOptions* request_options,
+    uint64_t correlation_id);
+
+/// Set the batch size for the request in a request options.
+/// \param request_options The request options object.
+/// \param batch_size The batch size.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestOptionsSetBatchSize(
+    TRTSERVER_InferenceRequestOptions* request_options, uint32_t batch_size);
+
+
+/// Add a input meta-data associated with the request in a request options.
+/// \param request_options The request options object.
+/// \param input_name The name of the input.
+/// \param dims The shape of the input. nullptr can be used if the input shape
+/// is fixed size.
+/// \param dim_count The length 'dims'.
+/// \param batch_byte_size The size of the full batch of the input tensor, in
+/// bytes. This field is only requried for non-fixed-size data type (like
+/// STRING). In the case of fixed-size data type, 0 can always be set for this
+/// field.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestOptionsAddInput(
+    TRTSERVER_InferenceRequestOptions* request_options, const char* input_name,
+    const int64_t* dims, uint64_t dim_count, uint64_t batch_byte_size);
+
+/// Request the raw data of the specified output to be returned for requests
+/// associated with the request options.
+/// \param request_options The request options object.
+/// \param output_name The name of the output.
+/// \return a TRTSERVER_Error indicating success or failure.
+TRTSERVER_EXPORT TRTSERVER_Error* TRTSERVER_InferenceRequestOptionsAddOutputRaw(
```

Comment:
To align better with the upcoming V2 API make this be "TRTSERVER_InferenceRequestOptionsAddOutput".